Dockerfile

```diff
@@ -0,0 +1,35 @@
+# Prerequisites for using dockerfile can be seen here https://github.com/zephyrtronium/robot/issues/76
+# docker build -t robot 
+# CONFIGDIR can be fed as an environment variable to set config directory
+
+# Main image
+FROM golang:1.23-alpine AS build
+
+# Reserve build argument
+ARG CONFIGDIR
+
+# Populate Go resources
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy repo
+COPY . .
+
+# Build Robot
+RUN --mount=type=cache,target=/root/.cache/go-build \
+  --mount=type=cache,target=/go/pkg \
+  CGO_ENABLED=0 \
+  go build -o /build/robot github.com/zephyrtronium/robot
+
+# Prepare minimised image
+FROM alpine
+
+# Set directory where config is read from, defaults to root in container
+ARG CONFIGDIR
+ENV CONFIGDIR=${CONFIGDIR:-/robot.toml}
+
+# Copy Robot binary
+COPY --from=build /build/robot /bin/robot
+
+# Provide Robot with config
+CMD ["sh", "-c", "/bin/robot -config $CONFIGDIR"]
```

Comment:
So I ran into issues trying to feed the variable into the container on startup

Turns out that variables can't be given directly without the `"sh", "-c"` because of how the runtime works

https://stackoverflow.com/a/40454758

I had to remove the entrypoint as well, as it wasn't happy with feeding in the variable after the entrypoint was set.

Right now if no variable is provided it will default to `/robot.toml` for the config directory, but if the environment variable is set when using `docker run` or `docker-compose` then you can change where the config is read from.

cc: @topi314 @zephyrtronium 

Ready for review again

Reply:
Adjusted